ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -940,6 +940,17 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
         goto error;
     }
 
+    /**
+     * Unfortunately the attempt to implement FI_MR_SCALABLE in the GNI provider
+     * doesn't work, at least not well.  Since we're asking for the 1.5 libfabric
+     * API now, we have to tell GNI we want to use Mr. Basic.  Using FI_MR_BASIC
+     * rather than FI_MR_VIRT_ADDR | FI_MR_ALLOCATED | FI_MR_PROV_KEY to stay
+     * compatible with older libfabrics.
+     */
+    if (!strcmp(prov->fabric_attr->prov_name,"gni")) {
```

Comment:
Do you care about any utility providers?

Reply:
The suggestion here would be to do a strncasecmp so it'll look at the leftmost provider for "gni" - future proofing for any layering that will happen on top.

Reply:
good idea I'll change to strncasecmp.
I'm nervous to make changes for other providers, hence the check only for gni.

Reply:
pushed change to strncmp